web3_proxy/src/stats/stat_buffer.rs

```diff
@@ -186,7 +187,8 @@ impl StatBuffer {
                         self.chain_id,
                         db_conn,
                         key,
-                        self.rpc_secret_key_cache.as_ref(),
+                        self.rpc_secret_key_cache.clone(),
+                        self.user_balance_cache.clone(), // Is cloning the Cache ok
```

Comment:
Cloning is cheap, but you can probably just pass by reference here

Reply:
looks like it's got to be a clone